examples/google-cloud-pubsub/lib/google/cloud/pubsub/v1.rb

```diff
@@ -0,0 +1,221 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/pubsub/v1/publisher_client"
+require "google/cloud/pubsub/v1/subscriber_client"
+
+module Google
+  module Cloud
+    module Pubsub
+      # rubocop:disable LineLength
+
+      ##
+      # # Ruby Client for Google Cloud Pub/Sub API ([Beta](https://github.com/GoogleCloudPlatform/google-cloud-ruby#versioning))
+      #
+      # [Google Cloud Pub/Sub API][Product Documentation]:
+      # Provides reliable, many-to-many, asynchronous messaging between
+      # applications.
+      # - [Product Documentation][]
+      #
+      # ## Quick Start
+      # In order to use this library, you first need to go through the following
+      # steps:
+      #
+      # 1. [Select or create a Cloud Platform project.](https://console.cloud.google.com/project)
+      # 2. [Enable billing for your project.](https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project)
+      # 3. [Enable the Google Cloud Pub/Sub API.](https://console.cloud.google.com/apis/library/pubsub.googleapis.com)
+      # 4. [Setup Authentication.](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/master/guides/authentication)
+      #
+      # ### Installation
+      # ```
+      # $ gem install google-cloud-pubsub
+      # ```
+      #
+      # ### Preview
+      # #### PublisherClient
+      # ```rb
+      # require "google/cloud/pubsub"
+      #
+      # publisher_client = Google::Cloud::Pubsub::Publisher.new(version: :v1)
+      # formatted_project = Google::Cloud::Pubsub::V1::PublisherClient.project_path(project_id)
```

Comment:
Are we going to add instance methods for all of these? If so, this should be simplified.